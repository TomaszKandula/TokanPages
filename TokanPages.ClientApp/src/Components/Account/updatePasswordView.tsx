import * as React from "react";
import Container from "@material-ui/core/Container";
import Grid from "@material-ui/core/Grid";
import Box from "@material-ui/core/Box";
import Typography from "@material-ui/core/Typography";
import Button from "@material-ui/core/Button";
import TextField from "@material-ui/core/TextField";
import { Card, CardContent, CircularProgress } from "@material-ui/core";
import { AccountCircle } from "@material-ui/icons";
import Skeleton from "@material-ui/lab/Skeleton";
import resetPasswordStyle from "./Styles/resetPasswordStyle";

interface IBinding 
{
    bind: IProperties;
}

interface IProperties
{
    isLoading: boolean;
    progress: boolean;
    caption: string;
    button: string;
    newPassword: string;
    verifyPassword: string;
    formHandler: any;
    buttonHandler: any;
    disableForm: boolean;
}

const UpdatePasswordView = (props: IBinding): JSX.Element =>
{
    const classes = resetPasswordStyle();
    return (
        <section>
            <Container maxWidth="sm">
                <Box pt={18} pb={10}>             
                    <Card elevation={4}>
                        <CardContent className={classes.card}>
                            <Box mb={3} textAlign="center">
                                <AccountCircle color="primary" style={{ fontSize: 72 }} />
                                <Typography variant="h5" component="h2" color="textSecondary">
                                    {props.bind?.isLoading ? <Skeleton variant="text" /> : props.bind?.caption}
                                </Typography>
                            </Box>
                            <Box>
                                <Grid container spacing={2}>
                                    <Grid item xs={12}>
                                        <TextField 
                                            onChange={props.bind?.formHandler} value={props.bind?.newPassword}  variant="outlined" required fullWidth 
                                            name="newPassword" id="newPassword" label="New password" autoComplete="password" disabled={props.bind?.disableForm}
                                        />
                                    </Grid>
                                    <Grid item xs={12}>
                                        <TextField 
                                            onChange={props.bind?.formHandler} value={props.bind?.verifyPassword}  variant="outlined" required fullWidth 
                                            name="verifyPassword" id="verifyPassword" label="Verify new password" autoComplete="password" disabled={props.bind?.disableForm}
                                        />
                                    </Grid>
                                </Grid>
                                <Box my={2}>
                                    <Button onClick={props.bind?.buttonHandler} type="submit" fullWidth variant="contained" color="primary" disabled={props.bind?.progress || props.bind?.disableForm}>
                                        {props.bind?.progress &&  <CircularProgress size={20} />}
                                        {!props.bind?.progress && props.bind?.button}
                                    </Button>
                                </Box>
                            </Box>
                        </CardContent>   
                    </Card>                    
                </Box>
            </Container>
        </section>
    );
}

export default UpdatePasswordView;
