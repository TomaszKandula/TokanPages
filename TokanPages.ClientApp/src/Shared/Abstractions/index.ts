export interface ViewProperties {
    isLoading: boolean;
}
