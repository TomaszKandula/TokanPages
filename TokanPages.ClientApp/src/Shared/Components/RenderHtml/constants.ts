export const BREAKING_WHITESPACE = " ";
export const NON_BREAKING_WHITESPACE = "&nbsp;";

export const BLOCK_TAG = ["div"];
export const TEXT_TAGS = ["p", "span", "b", "i", "u"];
export const QUOTE_TAGS = ["<blockquote>", "<q>", "<abbr>", "<address>", "<cite>"];
export const LINK_TAGS = ["a"];
export const LIST_TAGS = ["li", "ul", "ol"];
export const TABLE_TAGS = ["table", "th", "tr", "td"];
