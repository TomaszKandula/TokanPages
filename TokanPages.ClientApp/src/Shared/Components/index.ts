export { ApplicationDialogBox } from "./ApplicationDialogBox/applicationDialogBox";
export { ApplicationUserInfo } from "./ApplicationUserInfo/applicationUserInfo";
export { DocumentContentWrapper } from "./DocumentContentWrapper";
export { RenderContent } from "./RenderContent/renderContent";
export { RenderCardMedia } from "./CustomCardMedia/customCardMedia";
export { RenderImage } from "./CustomImage/customImage";
export { GetIcon } from "./GetIcon/getIcon";
export { RenderSideMenu } from "./ListRender/renderSideMenu";
export { RenderNavbarMenu } from "./ListRender/renderNavbarMenu";
export { ProgressBarView as ProgressBar } from "./ProgressBar";
export { ScrollToTop, HideOnScroll, ProgressOnScroll, ClearPageStart } from "./Scroll";
export { ApplicationToast } from "./Toasts/applicationToast";
export { UserAvatar } from "./UserAvatar";
export { UploadUserMedia } from "./UploadUserMedia";
export { TextFiedWithPassword } from "./TextFieldWithPassword";
export { ApplicationSession } from "./ApplicationSession";
export { BackArrow } from "./BackArrow";
export { ErrorBoundary } from "./ErrorBoundary";
