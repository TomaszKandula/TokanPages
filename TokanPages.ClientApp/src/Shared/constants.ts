import { IsRunningOnIOS } from "./Services/IsRunningOnIOS";

export const DELAY_3_SECONDS: number = 3000;
export const DELAY_30_SECONDS: number = 30000;
export const SET_INTERVAL_DELAY: number = IsRunningOnIOS() ? DELAY_3_SECONDS : DELAY_30_SECONDS;

export const LIKES_LIMIT_FOR_ANONYM: number = 25;
export const LIKES_LIMIT_FOR_USER: number = 50;
export const WORDS_PER_MINUTE: number = 265;
export const INTERNAL_SUBJECT_TEXT = "Incoming Business Inquiry";
export const INTERNAL_MESSAGE_TEXT = "Please check the internal payload for more details.";
export const RECEIVED_ERROR_MESSAGE: string = "RECEIVED_ERROR_MESSAGE";
export const NO_ERRORS: string = "NO_ERRORS";
export const USER_DATA: string = "USER_DATA";
export const PDF_JS_MIN_URL = "pdf.min.js";
export const PDF_WORKER_URL = "pdf.worker.min.js";
export const QUERY_SELECTOR = "meta[name=\"cache-content-info\"]";
export const META_ATTRIBUTE = "content";
export const PRERENDER_PATH_PREFIX = "/snapshot";
