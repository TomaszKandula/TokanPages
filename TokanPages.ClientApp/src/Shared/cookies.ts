import Validate from "validate.js";

interface ISetCookie 
{
    cookieName: string, 
    value: string, 
    days: number, 
    sameSite: string, 
    secure: string,
    exact?: string    
}

function SetCookie(props: ISetCookie): string
{
    let newCookie = "";
    let date = new Date();

    if (props.days)
    {

        let dateString = props.exact;

        if (Validate.isEmpty(props.exact))
        {
            date.setTime(date.getTime() + (props.days * 24 * 60 * 60 * 1000));
            dateString = date.toUTCString();
        }

        let LSecure = !Validate.isEmpty(props.secure) ? `; ${props.secure}` : "";
        newCookie = `${props.cookieName}=${props.value}; expires=${dateString}; path=/; SameSite=${props.sameSite} ${LSecure}`;

        document.cookie = newCookie;

    }

    return newCookie;
}

interface IGetCookie
{
    cookieName: string
}

function GetCookie(props: IGetCookie): string
{
    let cookieName = `${props.cookieName}=`;
    let cookieArray = document.cookie.split(";");

    for (let item of cookieArray)
    {
        let cookie = item;
        while (cookie.charAt(0) === " ")
        {
            cookie = cookie.substring(1, cookie.length);
        }

        if (cookie.indexOf(cookieName) === 0)
        {
            return cookie.substring(cookieName.length, cookie.length);
        }
    }

    return "";
}

interface IEraseCookie
{
    cookieName: string
}

function EraseCookie(props: IEraseCookie)
{
    document.cookie = `${props.cookieName}=; Path=/; Expires=Thu, 01 Jan 1970 00:00:01 GMT;`;
}

export 
{
    SetCookie,
    GetCookie,
    EraseCookie
}
