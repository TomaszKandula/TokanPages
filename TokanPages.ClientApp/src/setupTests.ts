// jest-dom adds custom jest matchers for asserting on DOM nodes.
// allows you to do things like:
// expect(element).toHaveTextContent(/react/i)
// learn more: https://github.com/testing-library/jest-dom
import "@testing-library/jest-dom";
import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import enableHooks from "jest-react-hooks-shallow";

enableHooks(jest);

jest.mock("react-redux", () => (
{
    useSelector: jest.fn(),
    useDispatch: jest.fn()
}));

jest.mock("connected-react-router", () => (
{
    connectRouter: jest.fn(),
    routerMiddleware: jest.fn()
}));

Enzyme.configure({ adapter: new Adapter(), disableLifecycleMethods: false });

// environmental variables used during all tests
process.env.REACT_APP_API_VER="1";
process.env.REACT_APP_BACKEND="http://localhost:5000";
