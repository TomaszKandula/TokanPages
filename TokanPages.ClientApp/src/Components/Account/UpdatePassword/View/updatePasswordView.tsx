import * as React from "react";
import Container from "@material-ui/core/Container";
import Grid from "@material-ui/core/Grid";
import Typography from "@material-ui/core/Typography";
import Button from "@material-ui/core/Button";
import { Card, CardContent, CircularProgress } from "@material-ui/core";
import { AccountCircle } from "@material-ui/icons";
import Skeleton from "@material-ui/lab/Skeleton";
import { ViewProperties } from "../../../../Shared/Abstractions";
import { ReactChangeEvent, ReactKeyboardEvent } from "../../../../Shared/types";
import { TextFiedWithPassword } from "../../../../Shared/Components";
import { UpdatePasswordProps } from "../updatePassword";

interface Properties extends ViewProperties, UpdatePasswordProps {
    progress: boolean;
    caption: string;
    button: string;
    newPassword: string;
    verifyPassword: string;
    keyHandler: (event: ReactKeyboardEvent) => void;
    formHandler: (event: ReactChangeEvent) => void;
    buttonHandler: () => void;
    disableForm: boolean;
    labelNewPassword: string;
    labelVerifyPassword: string;
}

const ActiveButton = (props: Properties): React.ReactElement => {
    return (
        <Button
            fullWidth
            type="submit"
            variant="contained"
            onClick={props.buttonHandler}
            className="button"
            disabled={props.progress || props.disableForm}
        >
            {!props.progress ? props.button : <CircularProgress size={20} />}
        </Button>
    );
};

export const UpdatePasswordView = (props: Properties): React.ReactElement => {
    return (
        <section className="section" style={props.background}>
            <Container className="container">
                <div style={{ paddingTop: props.pt ?? 96, paddingBottom: props.pb ?? 80 }}>
                    <Card elevation={0} className="card">
                        <CardContent className="card-content">
                            <div style={{ textAlign: "center", marginBottom: 24 }}>
                                <AccountCircle className="account" />
                                <Typography className="caption">
                                    {props.isLoading ? <Skeleton variant="text" /> : props.caption}
                                </Typography>
                            </div>
                            <div>
                                <Grid container spacing={2}>
                                    <Grid item xs={12}>
                                        {props.isLoading ? (
                                            <Skeleton variant="rect" width="100%" height="45px" />
                                        ) : (
                                            <TextFiedWithPassword
                                                uuid="newPassword"
                                                fullWidth={true}
                                                value={props.newPassword}
                                                label={props.labelNewPassword}
                                                onKeyUp={props.keyHandler}
                                                onChange={props.formHandler}
                                                isDisabled={props.disableForm || props.progress}
                                            />
                                        )}
                                    </Grid>
                                    <Grid item xs={12}>
                                        {props.isLoading ? (
                                            <Skeleton variant="rect" width="100%" height="45px" />
                                        ) : (
                                            <TextFiedWithPassword
                                                uuid="verifyPassword"
                                                fullWidth={true}
                                                value={props.verifyPassword}
                                                label={props.labelVerifyPassword}
                                                onKeyUp={props.keyHandler}
                                                onChange={props.formHandler}
                                                isDisabled={props.disableForm || props.progress}
                                            />
                                        )}
                                    </Grid>
                                </Grid>
                                <div style={{ marginTop: 16, marginBottom: 16 }}>
                                    {props.isLoading ? (
                                        <Skeleton variant="rect" width="100%" height="40px" />
                                    ) : (
                                        <ActiveButton {...props} />
                                    )}
                                </div>
                            </div>
                        </CardContent>
                    </Card>
                </div>
            </Container>
        </section>
    );
};
