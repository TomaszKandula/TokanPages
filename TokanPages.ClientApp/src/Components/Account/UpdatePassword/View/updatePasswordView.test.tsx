import "../../../../setupTests";
import React from "react";
import { shallow } from "enzyme";
import { UpdatePasswordView } from "./updatePasswordView";
import { UPDATE_FORM } from "../../../../Shared/constants";

describe("Test account group component: updatePasswordView.", () => 
{
    it("Renders correctly '<UpdatePasswordView />' when content is loaded.", () => 
    {
        const tree = shallow(<UpdatePasswordView bind=
        {{
            isLoading: false,
            progress: false,
            caption: UPDATE_FORM,
            button: "Submit",
            newPassword: "user1password",
            verifyPassword: "user1password",
            formHandler: jest.fn(),
            buttonHandler: jest.fn(),
            disableForm: false,
            labelNewPassword: "New password",
            labelVerifyPassword: "Verify password"
        }}/>);
        expect(tree).toMatchSnapshot();
    });
});
