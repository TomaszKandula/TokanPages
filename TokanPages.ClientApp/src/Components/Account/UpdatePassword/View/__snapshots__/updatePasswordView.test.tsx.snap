// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test account group component: updatePasswordView. Renders correctly '<UpdatePasswordView />' when content is loaded. 1`] = `
<section
  className="makeStyles-section-1"
>
  <WithStyles(ForwardRef(Container))
    maxWidth="sm"
  >
    <Styled(MuiBox)
      pb={10}
      pt={18}
    >
      <WithStyles(ForwardRef(Card))
        className="makeStyles-card-4"
        elevation={0}
      >
        <WithStyles(ForwardRef(CardContent))
          className="makeStyles-card_content-5"
        >
          <Styled(MuiBox)
            mb={3}
            textAlign="center"
          >
            <Memo(ForwardRef(AccountCircleIcon))
              className="makeStyles-account-2"
            />
            <WithStyles(ForwardRef(Typography))
              className="makeStyles-caption-3"
            >
              Update Password
            </WithStyles(ForwardRef(Typography))>
          </Styled(MuiBox)>
          <Styled(MuiBox)>
            <WithStyles(ForwardRef(Grid))
              container={true}
              spacing={2}
            >
              <WithStyles(ForwardRef(Grid))
                item={true}
                xs={12}
              >
                <WithStyles(ForwardRef(TextField))
                  autoComplete="password"
                  disabled={false}
                  fullWidth={true}
                  id="newPassword"
                  label="New password"
                  name="newPassword"
                  onChange={[MockFunction]}
                  onKeyUp={[MockFunction]}
                  required={true}
                  type="password"
                  value="user1password"
                  variant="outlined"
                />
              </WithStyles(ForwardRef(Grid))>
              <WithStyles(ForwardRef(Grid))
                item={true}
                xs={12}
              >
                <WithStyles(ForwardRef(TextField))
                  autoComplete="password"
                  disabled={false}
                  fullWidth={true}
                  id="verifyPassword"
                  label="Verify password"
                  name="verifyPassword"
                  onChange={[MockFunction]}
                  onKeyUp={[MockFunction]}
                  required={true}
                  type="password"
                  value="user1password"
                  variant="outlined"
                />
              </WithStyles(ForwardRef(Grid))>
            </WithStyles(ForwardRef(Grid))>
            <Styled(MuiBox)
              my={2}
            >
              <ActiveButton
                button="Submit"
                buttonHandler={[MockFunction]}
                caption="Update Password"
                disableForm={false}
                formHandler={[MockFunction]}
                isLoading={false}
                keyHandler={[MockFunction]}
                labelNewPassword="New password"
                labelVerifyPassword="Verify password"
                newPassword="user1password"
                progress={false}
                verifyPassword="user1password"
              />
            </Styled(MuiBox)>
          </Styled(MuiBox)>
        </WithStyles(ForwardRef(CardContent))>
      </WithStyles(ForwardRef(Card))>
    </Styled(MuiBox)>
  </WithStyles(ForwardRef(Container))>
</section>
`;
